aodndata/soop/soop_xbt_dm.py

```diff
@@ -42,42 +41,52 @@ def dest_path(filepath):
 
 def create_plot(netcdfFilePath, output_dir):
     """ create plot to be ingested in SOOP XBT NRT pipeline"""
-    F = Dataset(netcdfFilePath, 'r', format='NETCDF4')
-    cruise_id = F.XBT_cruise_ID
-    sea_water_temperature = F.variables['TEMP']
-    depth = F.variables['DEPTH']
-    time = F.variables['TIME']
-    time = num2date(time[:], time.units, time.calendar)
-    lat = F.variables['LATITUDE'][:]
-    lon = F.variables['LONGITUDE'][:]
-    xbt_unique_id = F.XBT_uniqueid
+    with Dataset(netcdfFilePath, 'r', format='NETCDF4') as F:
+        cruise_id = F.XBT_cruise_ID
+        sea_water_temperature = F.variables['TEMP']
+        depth = F.variables['DEPTH']
+        time = num2date(F.variables['TIME'][:], F.variables['TIME'].units, F.variables['TIME'].calendar)
+        lat = F.variables['LATITUDE'][:]
+        lon = F.variables['LONGITUDE'][:]
+        xbt_unique_id = F.XBT_uniqueid
+        title_dataset = F.title
+        xbt_line_description = F.XBT_line_description
+        temp_qc = F.variables['TEMP_quality_control'][:]
+        time_coverage_start = F.time_coverage_start
+        temp_values = sea_water_temperature[:]
+        depth_values = depth[:]
+        sst_fillvalue = sea_water_temperature._FillValue
+        sst_units = sea_water_temperature.units
+        sst_longname = sea_water_temperature.long_name
+        depth_units = depth.units
+        depth_longname = depth.long_name
+        if hasattr(depth, '_FillValue'):
```

Comment:
I'd do it like this, since querying the scope via locals() is a bit clunky:

```python
try:
    depth_fillvalue = depth._FillValue
except AttributeError:
    depth_fillvalue = None
```

Reply:
Even better